core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -135,6 +140,11 @@ public void setTermsLookupFetchService(@Nullable  TermsLookupFetchService termsL
         this.termsLookupFetchService = termsLookupFetchService;
     }
 
+    @Inject(optional = true)
```

Comment:
I looked at it quickly and I wonder if we could do this:

 * given that ShapeFetchService is only used in one place (the parser here) we could trash the service and copy it's code into the QueryParser
 * Add a Client reference to the `QueryShardContext` and get rid of this injector here. 

then we only need a client on the  `QueryParseContent` which might make way more sense since then we can trash  `TermsLookupFetchService` there too and be fully generic with a client. does this make sense?